packages/plugin-iteratively-node/lib/__tests__/smoke.test.ts

```diff
@@ -18,19 +41,298 @@ afterEach(() => {
   itly = undefined;
 });
 
-test('should not crash on load', () => {
+test.each([
+  ['production'],
+  ['development'],
+])('should not crash on load (env: %s)', (environment) => {
   const iterativelyPlugin = new IterativelyNodePlugin(
-    iterativelyApiKey,
+    defaultTestApiKey,
     {
-      disabled: true,
-      url: 'https://localhost:4000',
+      url: defaultTestUrl,
+      environment,
     },
   );
 
   expect(() => {
     itly.load({
-      environment: 'production',
+      environment,
       plugins: [iterativelyPlugin],
     });
   }).not.toThrow();
 });
+
+test('should not post if on production', () => {
+  const environment = 'production';
+
+  const iterativelyPlugin = new IterativelyNodePlugin(
+    defaultTestApiKey,
+    {
+      environment,
+      url: defaultTestUrl,
+      flushAt: 1,
+    },
+  );
+
+  itly.load({
+    environment,
+    plugins: [iterativelyPlugin],
+  });
+
+  itly.track(defaultUserId, defaultTestEvent);
+
+  expect(fetch).not.toBeCalled();
+});
+
+test('should post when flushAt reached', async () => {
+  const environment = 'development';
+
+  const iterativelyPlugin = new IterativelyNodePlugin(
+    defaultTestApiKey,
+    {
+      environment,
+      url: defaultTestUrl,
+      flushAt: 2,
+      flushInterval: 10000,
+    },
+  );
+
+  itly.load({
+    environment,
+    plugins: [iterativelyPlugin],
+  });
+
+  const events = [0, 1].map((i) => ({
+    ...defaultTestEvent,
+    properties: {
+      iteration: i,
+    },
+  }));
+
+  events.forEach((event) => itly.track(defaultUserId, event));
+
+  expect(fetch).toHaveBeenCalledTimes(1);
+  expect(fetch).toHaveBeenCalledWith(defaultTestUrl, defaultFetchRequest);
+
+  expect(JSON.parse((fetch as any).mock.calls[0][1].body)).toMatchObject({
+    objects: expect.arrayContaining(
+      events.map((event) => expect.objectContaining({
+        type: 'track',
+        eventId: event.id,
+        eventName: event.name,
+        eventSchemaVersion: event.version,
+        properties: event.properties,
+        valid: true,
+        validation: {
+          details: '',
+        },
+      })),
+    ),
+  });
+});
+
+test('should post in flushInterval', async () => {
+  const environment = 'development';
+  const flushInterval = 10;
+
+  const iterativelyPlugin = new IterativelyNodePlugin(
+    defaultTestApiKey,
+    {
+      environment,
+      url: defaultTestUrl,
+      flushAt: 100,
```

Comment:
Working on Android I found `flushAt` and `flushInterval` too ambiguous and renamed them `flushQueueSize` and `flushIntervalMillis`. What do you think?